File: internal/content/testdata/bench/typescript/maximum-subarray-sum.ts
function max_subarray_sum(nums: number[]): number {
    let best = nums[0];
    let current = nums[0];
    for (let i = 1; i < nums.length; i++) {
        current = Math.max(nums[i], current + nums[i]);
        best = Math.max(best, current);
    }
    return best;
}

File: internal/content/testdata/bench/java/maximum-subarray-sum.java
import java.util.*;

class Solution {
    public long max_subarray_sum(long[] nums) {
        long best = nums[0];
        long current = nums[0];
        for (int i = 1; i < nums.length; i++) {
            current = Math.max(nums[i], current + nums[i]);
            best = Math.max(best, current);
        }
        return best;
    }
}
